### PR TITLE
Add project updates to adminbro

### PR DESCRIPTION
**Pull request by CarlosQ96**
Some feedback from Ashley and Nikola.
Currently Verification team cant see the project updates.

### CODE REVIEW

**Comment by CarlosQ96** [comment]
This adds the project update to the project view and links it to its show page, also adds the sidebar project updates (I limit it for now only to show and list).
image

File: src/entities/project.ts [hunk 1]
```diff
@@ -299,6 +299,12 @@ class Project extends BaseEntity {
   @Field(type => ProjectUpdate, { nullable: true })
   projectUpdate?: any;
 
+  @Field(type => [ProjectUpdate], { nullable: true })
+  projectUpdates?: ProjectUpdate[];
+
+  @Field(type => String, { nullable: true })
+  adminBroBaseUrl: string;
```
Comment (CarlosQ96) [inline src/entities/project.ts:306]: I found no other way to pass the env var to the react view. So I added a virtual attribute I fill with the env var, and pass it as props.

File: src/server/adminBro.ts [hunk 2]
```diff
@@ -220,7 +226,9 @@ export const setSocialProfiles: After<ActionResponse> = async (
   const projectId = record.params.projectId || record.params.id;
 
   const socials = await findSocialProfilesByProjectId({ projectId });
+  const projectUpdates = await findProjectUpdatesByProjectId(projectId);
   const project = await findProjectById(projectId);
+  const adminBroBaseUrl = process.env.SERVER_URL;
```
Comment (CarlosQ96) [inline src/server/adminBro.ts:231]: Here is where I fill the react component with the env var value.

File: src/repositories/projectUpdateRepository.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { ProjectUpdate } from '../entities/project';
+
+export const findProjectUpdatesByProjectId = async (
+  projectId: number,
+): Promise<ProjectUpdate[] | []> => {
+  return ProjectUpdate.createQueryBuilder('projectUpdate')
+    .where('projectUpdate.projectId = :projectId', { projectId })
+    .andWhere('projectUpdate.isMain = true')
```
Comment (mohammadranjbarz) [inline src/repositories/projectUpdateRepository.ts]: @CarlosQ96  Why do we need this condition ? 
` .andWhere('projectUpdate.isMain = true')`

Comment (CarlosQ96) [reply src/repositories/projectUpdateRepository.ts]: Whoops I made  mistake here. It should be isMain = false.

Comment (CarlosQ96) [reply src/repositories/projectUpdateRepository.ts]: Default projectUpdate is a hidden project update to add reactions to, only.

Comment (CarlosQ96) [reply src/repositories/projectUpdateRepository.ts]: I think Adminbro shouldn't really show it in the project view, doesn't add much.

Comment (CarlosQ96) [reply src/repositories/projectUpdateRepository.ts]: Pushed the change.

Comment (mohammadranjbarz) [reply src/repositories/projectUpdateRepository.ts]: Yeah now makes sense, thank you

**Review by mohammadranjbarz — APPROVED** [review]
It seems good to me, thanks @CarlosQ96

**Comment by mohammadranjbarz** [comment]
@CarlosQ96  if we dont have an issue for that, you can ask @maryjaf here to  test it

**Comment by CarlosQ96** [comment]
Sure! Yeah we don't really have an issue. 
@maryjaf This is a minor issue, but whenever you can test it, it would be good. Quality of life for adminbro.

Basically the projectUpdates should be visible in the project Show view (when you click a specific project in adminbro).
It should have working links to the user and to the projectUpdate show view.

By default this added a link in the sidebar to the projectUpdate (listing all). Right now its not posible to hide. But I disabled from that view all actions, leaving only the option to filter and show page of a project update.

**Comment by CarlosQ96** [comment]
Because projectUpdate should not be modified by admins directly, thats why I disabled actions on them.

(there is another issue not yet implemented to review them, but that's not yet implemented) However this set the ground work so when we do that, it's much much easier.

**Comment by maryjaf** [comment]
image
I think I did not understand the problem correctly @CarlosQ96 
But currently, the following links are displayed on the project details page in the "Project Updates" section
Adminbro User Link
https://serve.giveth.io/admin/resources/User/records/379/show
Project Update Adminbro Edit Link
https://serve.giveth.io/admin/resources/ProjectUpdate/records/2474/show(As you said, now there is only a show and there is no editing feature)

**Comment by CarlosQ96** [comment]
@maryjaf The problem is:
Ashley told me they'd cant read project updates in the adminbro. So I added this section so they could.

This also was half the work for the next feature that will be allow admins to review project updates.
But I had to do this part first.

So think of it as the previous steps to the next feature haha.

**Pull request merged**